### PR TITLE
[TE] clean up some legacy code

### PR DESCRIPTION
**Pull request by akshayrai**
Changes:
* Deprecated and refactored the `getDimension` method in Merged Anomaly
* Clean up all the related and unused legacy classes

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/integration/AnomalyApplicationEndToEndTest.java [hunk 1]
```diff
@@ -1,435 +0,0 @@
-/**
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/integration/AnomalyApplicationEndToEndTest.java]: Why delete this test? Looks like it's still relevant.

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/detection/algorithm/LegacyAnomalyFunctionAlgorithmTest.java [hunk 2]
```diff
@@ -1,98 +0,0 @@
...
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
-import org.testng.Assert;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.Test;
-
-import static org.apache.pinot.thirdeye.dataframe.util.DataFrameUtils.*;
-
-
-public class LegacyAnomalyFunctionAlgorithmTest {
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/detection/algorithm/LegacyAnomalyFunctionAlgorithmTest.java]: Could you also delete the `LegacyAnomalyFuncitonAlgorithm` that is associated with the test? same with other classes.

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/detection/algorithm/LegacyAnomalyFunctionAlgorithmTest.java]: That class has many dependencies. I will remove it in a different PR.

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/integration/AnomalyApplicationEndToEndTest.java (hunk 1 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/integration/AnomalyApplicationEndToEndTest.java]: This test is still relying on the legacy classes like DetectionJobScheduler and AlertJobSchedulerV2. It needs to be rewritten.

**Pull request merged**